src/main/java/com/pingcap/tikv/codec/BytesUtils.java

```diff
@@ -52,11 +52,35 @@ public static void writeBytes(CodecDataOutput cdo, byte[] data) {
         }
     }
 
+    // WriteBytesDesc first encodes bytes using EncodeBytes, then bitwise reverses
+    // encoded value to guarantee the encoded value is in descending order for comparison.
+    public static void writeBytesDesc(CodecDataOutput cdo, byte[] data) {
+        writeBytes(cdo, data);
+        byte[] encodedData = cdo.toBytes();
+        // not sure how to empty cdo
```

Comment:
Please add unit test from now on. We need to take sometime to taking care of it for existing code. And for new classes, no tests no merge.
And please add hook to schema construction and test it on real tables. These doesn't need to be part of commit but it will serve a guideline for your unit tests.